lib/authCodeGrant.js

```diff
@@ -252,19 +252,19 @@ function redirect (done) {
   var url,
     state;
 
+  if(typeof this.req.body.state !== 'undefined') {
```

Comment:
Do we really need to check the types? I'm guessing we are expecting the state to be a string?
won't this work?

state = this.req.body.state || this.req.query.state;

(also do we need to encode/validate the state to prevent XSS?)